python/ray/data/_internal/execution/operators/map_operator.py

```diff
@@ -279,7 +279,7 @@ def _task_done_callback(task_index, inputs):
             metadata = [input[1] for input in inputs.blocks]
             locations = ray.experimental.get_object_locations(blocks)
             for block, meta in zip(blocks, metadata):
-                if locations[block]["did_spill"]:
+                if "did_spill" in locations[block]:
```

Comment:
Separate `KeyError` I ran into:
```
  File "/Users/michaelhly/Projects/ray/python/ray/data/_internal/execution/streaming_executor.py", line 187, in run
    while self._scheduling_loop_step(self._topology) and not self._shutdown:
  File "/Users/michaelhly/Projects/ray/python/ray/data/_internal/execution/streaming_executor.py", line 235, in _scheduling_loop_step
    process_completed_tasks(topology)
  File "/Users/michaelhly/Projects/ray/python/ray/data/_internal/execution/streaming_executor_state.py", line 333, in process_completed_tasks
    active_tasks[ref].on_waitable_ready()
  File "/Users/michaelhly/Projects/ray/python/ray/data/_internal/execution/interfaces/physical_operator.py", line 76, in on_waitable_ready
    self._task_done_callback()
  File "/Users/michaelhly/Projects/ray/python/ray/data/_internal/execution/operators/map_operator.py", line 299, in <lambda>
    lambda: _task_done_callback(task_index, inputs),
  File "/Users/michaelhly/Projects/ray/python/ray/data/_internal/execution/operators/map_operator.py", line 282, in _task_done_callback
    if locations[block]["did_spill"]:
KeyError: 'did_spill'
```

Reply:
we recently made changes to this code path, have you recently pulled master and rebuilt from source? are you still running into this issue?

Reply:
Yes. My tip is now at c399fcd7fccd3b3811c1c4f336505e1e6a2da9e0

Reply:
do you have a code sample you are running to get this error?

Reply:
Yes. Here is the script from the issue:
```python
import numpy as np
import ray
ds = ray.data.from_items([
    {"name": "Luna", "age": 4, "nicknames": ["Looney", "Loona"]},
    {"name": "Rory", "age": 14, "nicknames": ["Rorey"]},
    {"name": "Scout", "age": 9, "nicknames": ["Scoot"]},
])
def identity(batch):
    return batch
ds.map_batches(identity).map_batches(identity).show()
```

Reply:
Also I guess this check should actually be:
`if "did_spill" in locations[block] and locations[block]["did_spill"]:`

Reply:
Oh, may I need to run `pip install -e .` again?

Reply:
Okay. I forgot to rebuild the cpp changes from source. Reverting this change.

Reply:
Thanks, i'm able to reproduce this on my end as well. We are working internally to address it, thanks for reporting!